src/ofdm.c

```diff
@@ -1140,12 +1158,56 @@ static float est_timing_and_freq(struct OFDM *ofdm,
         complex float mvec[Npsam];
         for(int i=0; i<Npsam; i++) {
             complex float ph = cmplx(w*i);
-            mvec[i] = known_samples[i]*ph;
+            mvec[i] = conjf(known_samples[i]*ph);
         }
         for(int t=0; t<Ncorr; t+=tstep) {
+#if USE_VECTOR_OPS
+            float *rxPtr = (float*)&rx[t];
+            float *vecPtr = (float*)mvec;
+            float4 accumPos = { 0, 0, 0, 0 };
+            float4 accumNeg = { 0, 0, 0, 0 };
+            float4 accumImag = { 0, 0, 0, 0 };
+            float corrR = 0, corrI = 0;
+            int numBlocks = Npsam >> 1;
+            for (int i = 0; i < numBlocks; i++)
+            {
+                /* Lay out vectors as follows: 
+                   vec1 = rx[0].a, rx[0].b, rx[1].a, rx[1].b, ...
+                   vec2 = mvec[0].c, mvec[0].d, mvec[1].c, mvec1[1].d, ... */
+                float4 vec1 = { rxPtr[0], rxPtr[1], rxPtr[2], rxPtr[3] };
+                float4 vec2 = { vecPtr[0], vecPtr[1], vecPtr[2], vecPtr[3] };
+
+                accumPos += vec1 * vec2;
+                accumNeg -= vec1 * vec2;
+
+                /* Lay out vec3 as { rx[0].b, rx[0].a, rx[1].b, rx[0].b, ... }.
+                   Multiply vec3 by vec2 to get us bc, ad, bc, ad
+                   and add to second accumulator. */
+                float4 vec3 = { rxPtr[1], rxPtr[0], rxPtr[3], rxPtr[2] };
+                accumImag += vec3 * vec2;
```

Comment:
Could the ordering `rxPtr[1], rxPtr[0], rxPtr[3], rxPtr[2]` be done outside the inner loop, and just have `pvec4`?

Reply:
Tried using a mvecRev array outside the loop that already does the reversing and it didn't seem to make a difference.